printer_tray/report_xml_action.py

```diff
@@ -19,20 +19,21 @@
 #
 ##############################################################################
 
-from openerp.osv import orm, fields
+from openerp import models, fields, api
 
 
-class ReportXMLAction(orm.Model):
+class ReportXMLAction(models.Model):
     _inherit = 'printing.report.xml.action'
 
-    _columns = {
-        'printer_tray_id': fields.many2one(
-            'printing.tray', 'Paper Source',
-            domain="[('printer_id', '=', printer_id)]"),
-        }
+    printer_tray_id = fields.Many2one(
```

Comment:
No,  the tray field can be empty and means that cups will print on the default tray of the printer. It is preferable to leave the field empty rather than copy this value from the cups server, because if the default tray is changed on the cups server,  the tray stored in the records would no longer be correct.